mula/scheduler/models/queue.py

```diff
@@ -1,65 +1,24 @@
+import enum
 import uuid
 from datetime import datetime, timezone
 
 from pydantic import BaseModel, ConfigDict, Field
-from sqlalchemy import Column, DateTime, Integer, String
+from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, UniqueConstraint
 from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import relationship
 from sqlalchemy.sql import func
 
 from scheduler.utils import GUID
 
 from .base import Base
+from .task import Task, TaskStatus
 
+# Make PrioritizedItemStatus an enum of TaskStatus
+PrioritizedItemStatus = enum.Enum("PrioritizedItemStatus", TaskStatus.__members__)
 
-class PrioritizedItem(BaseModel):
-    """Representation of an queue.PrioritizedItem on the priority queue. Used
-    for unmarshalling of priority queue prioritized items to a JSON
-    representation.
-    """
 
-    model_config = ConfigDict(from_attributes=True)
-
-    id: uuid.UUID = Field(default_factory=uuid.uuid4)
-
-    scheduler_id: str | None = None
-
-    # A unique generated identifier for the object contained in data
-    hash: str | None = Field(None, max_length=32)
-
-    priority: int | None = 0
-
-    data: dict = Field(default_factory=dict)
-
-    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
-
-    modified_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
-
-
-class PrioritizedItemDB(Base):
-    __tablename__ = "items"
-
-    id = Column(GUID, primary_key=True)
-
-    scheduler_id = Column(String)
-
-    hash = Column(String(32), index=True)
-
-    priority = Column(Integer)
-
-    data = Column(JSONB, nullable=False)
-
-    created_at = Column(
-        DateTime(timezone=True),
-        nullable=False,
-        server_default=func.now(),
-    )
-
-    modified_at = Column(
-        DateTime(timezone=True),
-        nullable=False,
-        server_default=func.now(),
-        onupdate=func.now(),
-    )
+class PrioritizedItem(Task):
```

Comment:
To make it easier for development I've opted (for now) to subclass a `PrioritizedItem` from a `Task`. This can be refactored eventually to fully remove the `PrioritizedItem` class from the scheduler

Reply:
So there are no issues here also when trying to query PrioritizedItems?

Reply:
There were some un pushed changes. It should now be removed